--- libfreenect-9999.orig/CMakeLists.txt	2016-02-19 08:02:12.348609254 +0100
+++ libfreenect-9999/CMakeLists.txt	2016-02-19 08:03:52.328902875 +0100
@@ -31,6 +31,7 @@
 ######################################################################################
 
 cmake_minimum_required(VERSION 2.6)
+set(PYTHON_EXECUTABLE "python3")
 
 ######################################################################################
 # Project declaration and options
