--- src/arduino.cc/builder/test/helper_tools_downloader.go.orig	2016-03-14 22:40:25.937123887 -0400
+++ src/arduino.cc/builder/test/helper_tools_downloader.go	2016-03-14 22:49:35.922357747 -0400
@@ -38,7 +38,6 @@
 	"encoding/json"
 	"fmt"
 	"github.com/go-errors/errors"
-	"golang.org/x/codereview/patch"
 	"io"
 	"io/ioutil"
 	"net/http"
@@ -147,21 +146,6 @@
 }
 
 func patchFiles(t *testing.T) {
-	files, err := ioutil.ReadDir(PATCHES_FOLDER)
-	NoError(t, err)
-
-	for _, file := range files {
-		if filepath.Ext(file.Name()) == ".patch" {
-			data, err := ioutil.ReadFile(Abs(t, filepath.Join(PATCHES_FOLDER, file.Name())))
-			NoError(t, err)
-			patchSet, err := patch.Parse(data)
-			NoError(t, err)
-			operations, err := patchSet.Apply(ioutil.ReadFile)
-			for _, op := range operations {
-				utils.WriteFileBytes(op.Dst, op.Data)
-			}
-		}
-	}
 }
 
 func download(t *testing.T, cores, boardsManagerCores, boardsManagerRedBearCores []Core, tools, toolsMultipleVersions, boardsManagerTools, boardsManagerRFduinoTools []Tool, libraries []Library) {
