diff -ur pigpio-9999.orig/Makefile pigpio-9999/Makefile
--- pigpio-9999.orig/Makefile	2019-03-05 12:26:57.330266931 +0100
+++ pigpio-9999/Makefile	2019-03-05 17:15:42.301814015 +0100
@@ -1,15 +1,6 @@
 #
 # Set CROSS_PREFIX to prepend to all compiler tools at once for easier
 # cross-compilation.
-CROSS_PREFIX =
-CC           = $(CROSS_PREFIX)gcc
-AR           = $(CROSS_PREFIX)ar
-RANLIB       = $(CROSS_PREFIX)ranlib
-SIZE         = $(CROSS_PREFIX)size
-STRIP        = $(CROSS_PREFIX)strip
-SHLIB        = $(CC) -shared
-STRIPLIB     = $(STRIP) --strip-unneeded
-
 SOVERSION    = 1
 
 CFLAGS	+= -O3 -Wall -pthread
@@ -38,7 +29,7 @@
 bindir = $(exec_prefix)/bin
 includedir = $(prefix)/include
 libdir = $(prefix)/lib
-mandir = $(prefix)/man
+mandir = $(prefix)/share/man
 
 all:	$(ALL)
 
@@ -103,8 +94,6 @@
 	install -m 0755 pig2vcd                        $(DESTDIR)$(bindir)
 	install -m 0755 pigpiod                        $(DESTDIR)$(bindir)
 	install -m 0755 pigs                           $(DESTDIR)$(bindir)
-	if which python2; then python2 setup.py install $(PYINSTALLARGS); fi
-	if which python3; then python3 setup.py install $(PYINSTALLARGS); fi
 	install -m 0755 -d                             $(DESTDIR)$(mandir)/man1
 	install -m 0644 *.1                            $(DESTDIR)$(mandir)/man1
 	install -m 0755 -d                             $(DESTDIR)$(mandir)/man3
@@ -126,8 +115,6 @@
 	rm -f $(DESTDIR)$(bindir)/pig2vcd
 	rm -f $(DESTDIR)$(bindir)/pigpiod
 	rm -f $(DESTDIR)$(bindir)/pigs
-	if which python2; then python2 setup.py install $(PYINSTALLARGS) --record /tmp/pigpio >/dev/null; sed 's!^!$(DESTDIR)!' < /tmp/pigpio | xargs rm -f >/dev/null; fi
-	if which python3; then python3 setup.py install $(PYINSTALLARGS) --record /tmp/pigpio >/dev/null; sed 's!^!$(DESTDIR)!' < /tmp/pigpio | xargs rm -f >/dev/null; fi
 	rm -f $(DESTDIR)$(mandir)/man1/pig*.1
 	rm -f $(DESTDIR)$(mandir)/man1/libpigpio*.1
 	rm -f $(DESTDIR)$(mandir)/man3/pig*.3
