diff -ur slic3r-9999.orig/lib/Slic3r.pm slic3r-9999/lib/Slic3r.pm
--- slic3r-9999.orig/lib/Slic3r.pm	2017-05-08 09:38:37.478019637 +0200
+++ slic3r-9999/lib/Slic3r.pm	2017-05-08 09:41:32.370470244 +0200
@@ -42,7 +42,7 @@
 
 use FindBin;
 # Path to the images.
-my $varpath = decode_path($FindBin::Bin) . "/var";
+my $varpath = decode_path($FindBin::RealBin) . "/var";
 if ($^O eq 'darwin' && !-d $varpath) {
     $varpath = decode_path($FindBin::Bin) . "/../Resources/var";
 }
