diff -urN vzlogger-9999.orig/src/Config_Options.cpp vzlogger-9999/src/Config_Options.cpp
--- vzlogger-9999.orig/src/Config_Options.cpp	2020-06-05 23:21:00.861747724 +0200
+++ vzlogger-9999/src/Config_Options.cpp	2020-06-05 23:24:56.564808551 +0200
@@ -303,16 +303,16 @@
 		case 18:
 		case 23:
 			if (*p != '-')
-				return FALSE;
+				return false;
 			else
 				break;
 
 		default:
 			if (!isxdigit(*p))
-				return FALSE;
+				return false;
 			else
 				break;
 		}
 	}
-	return TRUE;
+	return true;
 }
diff -urN vzlogger-9999.orig/src/local.cpp vzlogger-9999/src/local.cpp
--- vzlogger-9999.orig/src/local.cpp	2020-06-05 23:21:00.861747724 +0200
+++ vzlogger-9999/src/local.cpp	2020-06-05 23:26:52.092348207 +0200
@@ -153,7 +153,7 @@
 
 			if (strcmp(url, "/") == 0) {
 				if (options.channel_index()) {
-					show_all = TRUE;
+					show_all = true;
 				} else {
 					json_exception = json_object_new_object();
 
diff -urN vzlogger-9999.orig/src/protocols/MeterFile.cpp vzlogger-9999/src/protocols/MeterFile.cpp
--- vzlogger-9999.orig/src/protocols/MeterFile.cpp	2020-06-05 23:21:00.861747724 +0200
+++ vzlogger-9999/src/protocols/MeterFile.cpp	2020-06-05 23:25:55.128575191 +0200
@@ -113,7 +113,7 @@
 	try {
 		_rewind = optlist.lookup_bool(options, "rewind");
 	} catch (vz::OptionNotFoundException &e) {
-		_rewind = FALSE; // do not rewind file by default
+		_rewind = false; // do not rewind file by default
 	} catch (vz::InvalidTypeException &e) {
 		print(log_alert, "Invalid type for 'rewind'", name().c_str());
 		throw;
